app/validators/status_pin_validator.rb

```diff
@@ -5,6 +5,6 @@ def validate(pin)
     pin.errors.add(:base, I18n.t('statuses.pin_errors.reblog')) if pin.status.reblog?
     pin.errors.add(:base, I18n.t('statuses.pin_errors.ownership')) if pin.account_id != pin.status.account_id
     pin.errors.add(:base, I18n.t('statuses.pin_errors.private')) unless %w(public unlisted).include?(pin.status.visibility)
-    pin.errors.add(:base, I18n.t('statuses.pin_errors.limit')) if pin.account.status_pins.count > 4 && pin.account.local?
+    pin.errors.add(:base, I18n.t('statuses.pin_errors.limit')) if pin.account.status_pins.count >= (ENV['MAX_PINNED_TOOTS'].to_i || 5) && pin.account.local?
```

Comment:
I would move that to a constant. I would also document it in the `.env.production.sample`. Please also do not change the default value.

Reply:
I've made these changes. Please let me know if you need me to fix anything else!